public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx

```diff
@@ -142,6 +149,14 @@ function AssociatedDetectors({ embeddable, closeFlyout, setMode }) {
             savedAugmentObjectsArr
           );
           setSelectedDetectors(curSelectedDetectors);
+          maxAssociatedCount = uiSettings.get(
+            PLUGIN_AUGMENTATION_MAX_OBJECTS_SETTING
+          );
+          if (maxAssociatedCount <= curSelectedDetectors.length) {
+            setAssociationLimitReached(true);
+          } else {
+            setAssociationLimitReached(false);
+          }
```

Comment:
We should do the same logic as done in `AddAnomalyDetector`, lines 254-258, where you fetch the actual current set of associated objects. The above check may be wrong if there is a bunch of alerting monitors already associated, for example

Reply:
fixed